airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,158 @@
+        )
+
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 2
+        assert all(("repository", "airbytehq/integration-test") in x.record.data.items() for x in actual_messages.records)
+
+    def test_full_refresh_with_pagination(self):
+        """Ensure pagination"""
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+                headers={"Link": '<https://api.github.com/repos/{}/events?page=2>; rel="next"'.format(_CONFIG.get("repositories")[0])},
+            ),
+        )
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100, "page": 2},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+            ),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 4
+
+    def test_incremental_read(self):
+        """Ensure incremental sync.
+        Stream `Events` is semi-incremental, so all request  will be performed and only new records will be extracted"""
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps(find_template("events", __file__)), 200),
+        )
+
+        source = SourceGithub()
+        actual_messages = read(
+            source,
+            config=_CONFIG,
+            catalog=_create_catalog(sync_mode=SyncMode.incremental),
+            state=StateBuilder()
+            .with_stream_state("events", {"airbytehq/integration-test": {"created_at": "2022-06-09T10:00:00Z"}})
+            .build(),
+        )
+        assert len(actual_messages.records) == 1
+
+    def test_read_with_error(self):
+        """Ensure read() method does not raise an Exception and log message with error is in output"""
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse('{"message":"some_error_message"}', 403),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 0
```

Comment:
I think a very interesting behavior when there are errors is the stream status that we end up with. The fact that there are records or not seems less relevant to me.

Reply:
added check for stream_status